pages/manage/productions/index.test.js

```diff
@@ -1,58 +1,62 @@
+import { screen } from '@testing-library/vue';
+import userEvent from '@testing-library/user-event';
+
 import Index from './Index';
-import Vue from 'vue';
 
-import { mountWithEnvironment } from '../../../test/mountWithEnvironment';
+import { renderWithEnvironment } from '../../../test/renderWithEnvironment';
+
+beforeEach(() => {
+  renderWithEnvironment(Index);
+});
 
 test('returns full list when entering the page', async () => {
-  const wrapper = mountWithEnvironment(Index);
+  const productions = await screen.findAllByRole('listitem', {
+    name: 'production',
+  });
 
-  await Vue.nextTick();
+  expect(productions).toHaveLength(2);
 
-  const listItems = wrapper.findAll('.productions .list-group-item');
-  expect(listItems).toHaveLength(2);
+  const anotherOne = await screen.findByText('Another one');
+  const testing = await screen.findByText('Testing');
+
+  expect(anotherOne).toBeInTheDocument();
+  expect(testing).toBeInTheDocument();
 });
 
 test('returns a found item when searching with an existing production name', async () => {
-  const wrapper = mountWithEnvironment(Index);
+  const input = screen.getByRole('textbox');
+  await userEvent.type(input, 'Testing');
+
+  const productions = await screen.findAllByRole('listitem', {
+    name: 'production',
+  });
 
-  const searchInput = wrapper.find('#productions-overview-search');
+  expect(productions).toHaveLength(1);
 
-  await searchInput.setValue('Testing');
-  await Vue.nextTick();
+  const testing = await screen.findByText('Testing');
 
-  const listItems = wrapper.findAll('.productions .list-group-item');
-  expect(listItems).toHaveLength(1);
+  expect(testing).toBeInTheDocument();
 });
 
 test('returns no results when searching a wrong production name', async () => {
-  const wrapper = mountWithEnvironment(Index);
+  const input = screen.getByRole('textbox');
+  await userEvent.type(input, 'random search');
 
-  const searchInput = wrapper.find('#productions-overview-search');
-
-  await searchInput.setValue('test');
-  await Vue.nextTick();
-
-  const productionsContainer = wrapper.find('.productions-events-container');
-
-  const listItems = wrapper.findAll('.productions .list-group-item');
-  expect(listItems).toHaveLength(0);
-
-  expect(productionsContainer.text()).toContain(
+  const noProductions = await screen.findByText(
     'productions.overview.no_productions',
   );
+
+  expect(noProductions).toBeInTheDocument();
 });
 
 test('Enable remove button when selecting event', async () => {
-  const wrapper = mountWithEnvironment(Index);
+  const checkbox = screen.getByRole('checkbox', { name: 'eee' });
 
-  await Vue.nextTick();
-  await Vue.nextTick();
+  const removeButton = await screen.findByText('productions.overview.delete');
```

Comment:
Just to clarify, this `productions.overview.delete` is the actual text that's rendered right? I'm guessing because we have no translations/default messages in the tests? Just want to make sure I understand it right.

Reply:
Yes we mock the $t function

Reply:
Too bad that it makes the test less readable, but makes sense. 👍

Reply:
What would you suggest, using the actual translation values? The translation keys are most likely to stay no?

Reply:
Nah it's fine, it was mostly to make sure I understood it right so I know what "text" to look for when I write tests myself 😄 In another project I worked on (React) we used a translation package that also made it possible to set a default message whenever you called it with a key. So if the key doesn't exist, it would print the default message. We used that in the tests because it wouldn't change either (since the defaults never changed in the code, but would be overwritten in the translations if someone needed to change the exact copy). It's technically more or less the same, just a bit more readable in the tests, that's all. But I don't know if that's even possible with `vue-i18n` / `nuxt-i18n`. So this is definitely okay as well for me! 👍